source/Halibut/Transport/PollingClient.cs

```diff
@@ -95,5 +117,56 @@ void ExecutePollingLoop(object ignored)
                 }
             }
         }
+
+        async Task ExecutePollingLoopAsyncCatchingExceptions()
+        {
+            try
+            {
+                await ExecutePollingLoopAsync();
+            }
+            catch (Exception)
+            {
+                // We may get errors about the workingCancellationTokenSource being used when it is disposed, we don't care about that.
+            }
+        }
+        async Task ExecutePollingLoopAsync()
+        {
+            var retry = createRetryPolicy();
+            var sleepFor = TimeSpan.Zero;
+            while (!workingCancellationTokenSource.Token.IsCancellationRequested)
+            {
+                try
+                {
+                    try
+                    {
+                        retry.Try();
+                        await secureClient.ExecuteTransactionAsync(async (protocol, cancellationToken) =>
+                        {
+                            // We have successfully connected at this point so reset the retry policy
+                            // Subsequent connection issues will try and reconnect quickly and then back-off
+                            retry.Success();
+                            await protocol.ExchangeAsSubscriberAsync(subscription, handleIncomingRequest, int.MaxValue, cancellationToken);
+                        }, workingCancellationTokenSource.Token);
```

Comment:
Use the above mentioned token here (and line 167) instead of `workingCancellationTokenSource.Token` 🙂